trio/_core/_run.py

```diff
@@ -63,6 +64,10 @@
 # Used to log exceptions in instruments
 INSTRUMENT_LOGGER = logging.getLogger("trio.abc.Instrument")
 
+# Private flag to sort tasks before shuffling them, so that seeding
+# the Random instance `_r` can make scheduling deterministic.
+_ALLOW_DETERMINISTIC_SCHEDULING = False
```

Comment:
Let's record a few more details here to help us figure it out again later. How about moving the definition of `_r` next to this, and the comment should say something like: "When running under hypothesis, we want scheduling to be as deterministic as possible. pytest-trio's hypothesis integration monkeypatches this variable to True, and seeds the `_r` instance, which together should be be enough to eliminate scheduling as a source of non-determinism."